### PR TITLE
Update frontend dependencies

### LINKED ISSUE #459: Update FE dependencies
Many dependencies on the frontend (and resolutions) are very old, ~2018. We should have a proper upgrade pass and see how much of the current `package.json` can be upgraded without code changes. 
Ideally we should get rid of the `"resolutions"` section in our `package.json` file.

### PR DESCRIPTION
**Pull request by isSerge**
This PR includes the following changes:
- Update most npm packages to up-to-date versions
- Remove `resolutions` from `package.json`
- Replace `tslint` with `eslint` (I tried to keep the original rules)

I also removed namespace blocks from modules with class components because I experienced reference errors due to namespaces and components having same name, and removed leftover types and interfaces that are no longer used. 

I hope this should close #459

### CODE REVIEW

File: frontend/tsconfig.json
Comment (TarikGul) [inline frontend/tsconfig.json:32]: Is there a reason for this change? Im assuming it actually just simplifies the complifier config without changing anything? As in Css files are ignored, and svgs are ignored.

**Review by TarikGul — APPROVED** [review]
LGTM, thanks so much for the contribution.

Side question, why were the SVG's changed?

**Comment by TarikGul** [comment]
Also invloving the error, looks like the `.nvmrc` need to be updated to 14, as well as the CI node engine.

File: frontend/tsconfig.json
Comment (isSerge) [reply frontend/tsconfig.json:32]: yes, just simplification

**Comment by isSerge** [comment]
@TarikGul Regarding SVGs I had this error:
`SyntaxError: unknown: Namespace tags are not supported by default. React's JSX doesn't support namespace tags. You can turn on the 'throwIfNamespace' flag to bypass this warning.`
Probably this can be adjusted somewhere in the Webpack config, but since we're using CRA, it's not exposed unless we do `eject`

**Review by niklasad1 — APPROVED** [review]
nice work LGTM except that the CI fails.

**Comment by isSerge** [comment]
@niklasad1 it seems I need additional approval from **paritytech/ci** since this is my first contribution to this repository, how can we push this forward?

**Comment by niklasad1** [comment]
@isSerge it's fine these CI changes are trivial we just need 2 approvals which we got but would be if @jsdw could have look before merging :)

**Review by sergejparity — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by jsdw — APPROVED** [review]
Thanks for the contribution; looks great offhand!

We can test it more thoroughly when it merges so I'm up for that.

**Pull request merged**
